### PR TITLE
Make batch write support spark 2.4

### PR DESCRIPTION
**Pull request by birdstorm**
Also fix CI when trying to enable partition table read.

### CODE REVIEW

File: tikv-client/src/main/java/com/pingcap/tikv/catalog/Catalog.java [hunk 1]
```diff
@@ -218,6 +218,13 @@ public TiTableInfo getTable(TiDBInfo database, String tableName) {
     Objects.requireNonNull(database, "database is null");
     Objects.requireNonNull(tableName, "tableName is null");
     TiTableInfo table = metaCache.getTable(database, tableName);
+
+    if (table == null) {
```
Comment (marsishandsome) [inline tikv-client/src/main/java/com/pingcap/tikv/catalog/Catalog.java]: should we also do `reloadCache` when database is null?

Comment (birdstorm) [reply tikv-client/src/main/java/com/pingcap/tikv/catalog/Catalog.java]: done.

**Comment by birdstorm** [comment]
/run-all-tests tidb=release-2.1 tikv=release-2.1 pd=release-2.1

**Comment by birdstorm** [comment]
/run-all-tests tidb=release-2.1 tikv=release-2.1 pd=release-2.1

**Pull request merged**